docs/content/users/usage/troubleshooting.md

```diff
@@ -241,6 +241,23 @@ The error messages you get will be more informative than messages that come when
 
 You can also see the full Docker build using `~/.ddev/bin/docker-compose -f .ddev/.ddev-docker-compose-full.yaml build --no-cache --progress=plain`.
 
+### Disable Xdebug When Running PHP Applications like Composer or Terminus
+
+During the `docker build` process Xdebug is enabled and configured. If you have an IDE open and listening for connections, `ddev start` can hang. Always disable Xdebug in each `RUN` command:
+
+For commands that need to run as root:
+
+```dockerfile
+# Example only - ddev already runs self-update for you.
+RUN export XDEBUG_MODE=off; composer self-update
```

Comment:
I think this would be cleaner...

```suggestion
RUN XDEBUG_MODE=off composer self-update
```

Or how about `ENV XDEBUG_MODE=off` - and why isn't that already in DDEV? Why would we have the extension even enabled at build time?